### PR TITLE
Adding EDB docs and methods for available base configs

### LINKED ISSUE #279: EDB needs API and CLI to list valid 'bases'
Issue was brought about during review of #274. With the addition of several new 'base' configurations to the 'bootstrap' database, it is a little obscure what 'base' a user should call and when. Thus, we need some methods created to display and describe the 'bases' loaded in the database and when/why they are used. 

- [x] Add a list_bases() function to ElectrolyteDB
- [x] Add a CLI hook, e.g. "edb info --bases"
- [x] Documentation on what each base does and why

### PR DESCRIPTION
**Pull request by aladshaw3**
## Fixes/Addresses:

#279 

## Summary/Motivation:

Base configs have a number of options available and those options are not yet well documented. Adding methods to display options from EDB as well as documentation providing explanation of options. 

## Changes proposed in this PR:
- Add list_bases function to db_api
- Add list_bases CLI hook
- Add documentation on available options
- Update existing edb tests

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by srikanthallu — DISMISSED** [review]

**Review by TimBartholomew — DISMISSED** [review]
LGTM

**Comment by aladshaw3** [comment]
Waiting for review from @dangunter

File: docs/technical_reference/edb/index.rst [hunk 1]
```diff
@@ -28,7 +28,7 @@ The EDB is intended to support some known workflows out of the box, with lower-l
 are not sufficient.
 
 .. note::
```
Comment (dangunter) [inline docs/technical_reference/edb/index.rst]: I think we should remove this note.

File: docs/technical_reference/edb/index.rst [hunk 2]
```diff
@@ -9,17 +9,17 @@ Electrolyte Database (EDB)
...
+system is extensible to new use-cases. The native storage format for MongoDB is a `JSON <https://json.org>`_ document,
+and the expected structure and fields of the *component* and *reaction* data is defined by a
 `JSON Schema <https://json-schema.org>`_. Validation using those schemas is built into the API (though it can be disabled).
 
 To interface with the `IDAES Core Modeling Framework <https://idaes-pse.readthedocs.io/en/stable/user_guide/concepts.html>`_
-(IDAES-CMF, which underlies WaterTAP), add components and reactions to a "base" object and fetch the result as a Python 
-`dict`. This result can be used to configure and build IDAES objects (`ParameterBlocks`, `ReactionBlocks`, etc.). 
+(IDAES-CMF, which underlies WaterTAP), add components and reactions to a "base" object and fetch the result as a Python
+`dict`. This result can be used to configure and build IDAES objects (`ParameterBlocks`, `ReactionBlocks`, etc.).
 The API also has methods to construct component and reaction objects from IDAES configurations.
```
Comment (dangunter) [inline docs/technical_reference/edb/index.rst:24]: I would add a link to the "how-to" docs (and examples, if there are any, I forget) here

File: watertap/edb/commands.py [hunk 3]
```diff
@@ -369,3 +369,55 @@ def schema(output_file, output_format, data_type, url, database):
...
+    "-u", "--url", help="Database connection URL", default=ElectrolyteDB.DEFAULT_URL
+)
+@click.option(
+    "-d", "--database", help="Database name", default=ElectrolyteDB.DEFAULT_DB
+)
+def info(data_type, url, database):
+    print_messages = _log.isEnabledFor(logging.ERROR)
+
+    _log.info(f"Connecting to MongoDB at: {url}/{database}")
+    db = ElectrolyteDB(url=url, db=database)
```
Comment (dangunter) [inline watertap/edb/commands.py]: should catch `pymongo.errors.ConnectionFailure`

File: watertap/edb/db_api.py [hunk 4]
```diff
@@ -344,6 +344,42 @@ def get_base(self, name: str = None) -> Union[Result, Base]:
...
+            name: Name of the base to describe
+        Returns:
+            desc: String of the description of the base 
+        """
+        if name == "default_thermo":
+            desc = "ThermoConfig: Default uses FTPx state vars for Liq phase"
+        elif name == "reaction":
+            desc = "ReactionConfig: Blank reaction template"
+        else:
+            items = name.split("_")
```
Comment (dangunter) [inline watertap/edb/db_api.py:372]: should check that the number of items is at least 2

**Review by dangunter — CHANGES_REQUESTED** [review]
I made a few comments. I think we should also document the convention for the base names somewhere, perhaps in the schema (there is a "description" available for any/all fields that we could use) and/or some of the code that accesses it.

**Review by dangunter — APPROVED** [review]
Comments addressed, looks good.

**Review by adam-a-a — APPROVED** [review]
LGTM

**Pull request merged**